src/satosa/backends/saml2.py

```diff
@@ -139,6 +140,13 @@ def authn_request(self, context, entity_id):
                            exc_info=True)
             raise SATOSAAuthenticationError(context.state, "Failed to construct the AuthnRequest") from exc
 
+        if self.sp.config.allow_unsolicited is False:
```

Comment:
oh noes! `sp.config.allow_unsolicited` does not set the correct value. 
The correct value lies under `self.sp.config._sp_allow_unsolicited`..